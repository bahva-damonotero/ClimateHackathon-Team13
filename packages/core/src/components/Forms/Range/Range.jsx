import React from 'react';
import PropTypes from 'prop-types';

import { FormError, Label } from '..';

export default function Range({
  children,
  disabled,
  error,
  id,
  min,
  max,
  onChange,
  step,
  value,
  ...props
}) {
  return (
    <>
      <Label error={error} target={id}>
        {children}
      </Label>

      <FormError error={error} />

      <input
        id={id}
        className="usa-range"
        disabled={disabled}
        type="range"
        min={min}
        max={max}
        onChange={onChange}
        step={step}
        value={value}
        aria-valuemin={min}
        aria-valuemax={max}
        aria-valuenow={value}
        role="slider"
        {...props}
      />
    </>
  );
}

Range.propTypes = {
  /** Label for input */
  children: PropTypes.node.isRequired,
  /** Should the input be disabled? */
  disabled: PropTypes.bool,
  /** Error message to display */
  error: PropTypes.string,
  /** ID of the input */
  id: PropTypes.string.isRequired,
  /** Maximum value allowed */
  max: PropTypes.number.isRequired,
  /** Minimum value allowed */
  min: PropTypes.number.isRequired,
  /** Function called when the input is modified */
  onChange: PropTypes.func,
  /** Specifies the legal number intervals */
  step: PropTypes.number,
  /** Used to set the range value */
  value: PropTypes.number
};

Range.defaultProps = {
  disabled: false,
  step: 1
};
