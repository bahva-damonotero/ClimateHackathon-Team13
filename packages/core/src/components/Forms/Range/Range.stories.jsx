import React from 'react';

import Range from '.';
import { Form } from '..';

export default {
  title: 'Forms/Range',
  component: Range
};

const Template = (args) => (
  <Form>
    <Range {...args} />
  </Form>
);

const defaultArgs = {
  children: 'Label',
  id: 'range-example',
  min: 0,
  max: 100,
  step: 10
};

export const def = Template.bind({});
def.storyName = 'Default';
def.args = {
  ...defaultArgs
};

export const selection = Template.bind({});
selection.args = {
  ...defaultArgs,
  readOnly: true,
  value: 50
};

export const disabled = Template.bind({});
disabled.args = {
  ...defaultArgs,
  disabled: true
};

export const error = Template.bind({});
error.args = {
  ...defaultArgs,
  error: 'Error with form field'
};
