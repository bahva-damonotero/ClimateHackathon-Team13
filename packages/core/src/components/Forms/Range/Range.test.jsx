import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Range from '.';

describe('Range', () => {
  const id = 'range-example';
  const label = 'Range label';

  const RANGE_SELECTOR = `#${id}.usa-range`;
  const LABEL_SELECTOR = '.usa-label';

  const jsx = (
    <Range id={id} max={100} min={0}>
      {label}
    </Range>
  );

  it('should render correctly', () => {
    const { container } = render(jsx);

    expect(container.querySelector(RANGE_SELECTOR)).toBeTruthy();

    const labelNode = container.querySelector(LABEL_SELECTOR);
    expect(labelNode.innerHTML).toBe(label);
    expect(labelNode.getAttribute('for')).toBe(id);
  });

  it('should match snapshot', () => {
    const tree = renderer.create(jsx).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
