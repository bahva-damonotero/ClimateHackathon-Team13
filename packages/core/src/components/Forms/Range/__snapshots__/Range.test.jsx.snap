// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Range should match snapshot 1`] = `
Array [
  <label
    className="usa-label"
    htmlFor="range-example"
  >
    Range label
  </label>,
  <input
    aria-valuemax={100}
    aria-valuemin={0}
    className="usa-range"
    disabled={false}
    id="range-example"
    max={100}
    min={0}
    role="slider"
    step={1}
    type="range"
  />,
]
`;
